### PR TITLE
[🧪] NT-887 Campaign details experiment on Project page

### PR DESCRIPTION
**Pull request by eoji**
# 📲 What
Displaying `native_project_page_campaign_details` variants on Project screen.

# 🤔 Why
🧫

# 🛠 How
## `ProjectViewModel`
- `projectAndNativeCheckoutEnabled` output is now `projectDataAndNativeCheckoutEnabled` to pass `ProjectData` to `ProjectAdapter`
- `ProjectAdapter` now takes in a `ProjectData` instead of a `Project`

## `ProjectHolderViewModel`
- Added `blurbVariantIsVisible` output that emits a `Boolean` determining if we should be showing the variant blurb or the control one.
- Added `setBlurbTextViews` helper method to set both blurbs `TextView`s
- Clicking on `read_more` button opens Project campaign in `WebView`
- `ProjectHolderViewModel` is now configured with `ProjectData`
- Added new `View` group for blurb variant.

## tests
- `MockExperimentsClientType` now takes in a default variant and uses it to return when `variant` is called
- Updated tests in `ProjectViewModelTest` and added tests in `ProjectHolderViewModelTest`

# 👀 See
| Variant 1 🦋 | Variant 2 🦋 | Control 🐛 |
| --- | --- | ---|
| screenshot-2020-02-18_150417 | screenshot-2020-02-18_150417 | screenshot-2020-02-18_153711 |

# 📋 QA
Get yourself into a variant and take a look at a project.

# Story 📖
[NT-887]

[NT-887]: https://kickstarter.atlassian.net/browse/NT-887

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt [hunk 1]
```diff
@@ -484,11 +484,11 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>(), CancelPledge
...
     }
 
-    private fun renderProject(projectAndNativeCheckoutEnabled: Pair<Project, Boolean>) {
-        val project = projectAndNativeCheckoutEnabled.first
-        val nativeCheckoutEnabled = projectAndNativeCheckoutEnabled.second
-        this.adapter.takeProject(project, nativeCheckoutEnabled)
-        ProjectViewUtils.setActionButton(project, this.back_project_button, this.manage_pledge_button, this.view_pledge_button)
+    private fun renderProject(projectDataAndNativeCheckoutEnabled: Pair<ProjectData, Boolean>) {
+        val projectData = projectDataAndNativeCheckoutEnabled.first
+        val nativeCheckoutEnabled = projectDataAndNativeCheckoutEnabled.second
```
Comment (justinswart) [inline app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt]: Is this still feature-flagged on Android? We've removed old checkout completely on iOS.

File: app/src/main/java/com/kickstarter/ui/viewholders/ProjectViewHolder.java [hunk 2]
```diff
@@ -169,7 +174,12 @@ public ProjectViewHolder(final @NonNull View view, final @NonNull Delegate deleg
     this.viewModel.outputs.blurbTextViewText()
       .compose(bindToLifecycle())
       .compose(observeForUI())
-      .subscribe(blurb -> this.blurbTextView.setText(Html.fromHtml(TextUtils.htmlEncode(blurb))));
+      .subscribe(this::setBlurbTextViews);
```
Comment (justinswart) [inline app/src/main/java/com/kickstarter/ui/viewholders/ProjectViewHolder.java]: What is the `blurbVariantTextView` for if we're only experimenting on the button?

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt (hunk 1 above)
Comment (eoji) [reply app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt]: Yes 😢 but I have a ticket in JIRA in the backlog to remove it!

File: app/src/main/java/com/kickstarter/ui/viewholders/ProjectViewHolder.java (hunk 2 above)
Comment (eoji) [reply app/src/main/java/com/kickstarter/ui/viewholders/ProjectViewHolder.java]: We are but there are 2 blurb containers. `blurb_view_variant` with the normal blurb and read more arrow (the whole container is clickable) and `blurb_view_variant` where **only** the button is clickable

**Pull request merged**